activestorage/lib/active_storage/service/disk_service.rb

```diff
@@ -48,16 +48,24 @@ def delete(key)
       end
     end
 
+    def delete_prefixed(prefix)
+      instrument :delete_prefixed, prefix: prefix do
+        Dir.glob(path_for("#{prefix}*")).each do |path|
```

Comment:
`Dir.glob` can potentially return `.` and `..`. Should we do something specific here to filter them out? Or are they always filtered out via the prefix? (What happens if there's a blank prefix?)

Reply:
> `Dir.glob` can potentially return `.` and `..`.

I was under the impression it would only do so if we set the `FNM_DOTMATCH` flag:

```ruby
irb(main):006:0> Dir.glob("/Users/george/Desktop/*")
=> []
irb(main):007:0> Dir.glob("/Users/george/Desktop/*", File::Constants::FNM_DOTMATCH)
=> ["/Users/george/Desktop/.", "/Users/george/Desktop/..", "/Users/george/Desktop/.DS_Store", "/Users/george/Desktop/.localized"]
```

Is that platform-dependent?

> Or are they always filtered out via the prefix?

In Active Storage’s own usage, yes.

> What happens if there's a blank prefix?

All objects are deleted. That seems *correct*, though perhaps not desirable.

Reply:
> Is that platform-dependent?

Not sure. It could be Ruby version dependent. I definitely remember seeing it in our app on Ruby 2.2.

> All objects are deleted. That seems correct, though perhaps not desirable.

I meant to say: if `.` and `..` are included and the prefix is blank would it delete the current and parent directories?

Reply:
> In Active Storage’s own usage, yes.

I’m wrong. Active Storage’s usage of `delete_prefixed` would theoretically encounter this.

> Not sure. It could be Ruby version dependent. I definitely remember seeing it in our app on Ruby 2.2.

I’m not seeing it on Ruby 2.2:

```ruby
irb(main):002:0> RUBY_VERSION
=> "2.2.8"
irb(main):003:0> Dir.glob("/Users/george/Desktop/*")
=> []
irb(main):004:0> Dir.glob("/Users/george/Desktop/*", File::Constants::FNM_DOTMATCH)
=> ["/Users/george/Desktop/.", "/Users/george/Desktop/..", "/Users/george/Desktop/.DS_Store", "/Users/george/Desktop/.localized"]
```

> I meant to say: if `.` and `..` are included and the prefix is blank would it delete the current and parent directories?

If they’re returned by `Dir.glob`, yes.

Reply:
Ah, I was thinking of `Dir.entries`. No worries here then 👍